core/src/main/java/com/squarespace/template/plugins/platform/ContentFormatters.java

```diff
@@ -378,22 +379,27 @@ public void apply(Context ctx, Arguments args, Variables variables) throws CodeE
     }
 
     private String getAltText(Context ctx) {
-      // For image blocks, caption is stored on the block and not the item.
-      // need to reach out via the context to see if it exist first,
-      // before falling back on the data on the item
-
-      // this will be empty if this is not a block
-      JsonNode blockInfo = ctx.resolve("info");
-      if (blockInfo != null) {
-        String altText = StringUtils.trimToNull(blockInfo.path("altText").asText());
-        if (altText != null) {
-          return altText;
-        }
+      // Content Items for image blocks were populated with an altText value with a migration. See CMS-33805.
+      // The contentItem value should always be used, even if it is empty.
+      JsonNode blockType = ctx.resolve("blockType");
+
```

Comment:
Not sure why we need to check for `blockType` here. Couldn't you just check for existence on the `altText` node and always return its value when it exists?
```
JsonNode altText = ctx.node().path("altText");
if (!altText.isMissingNode()) {
  return StringUtils.trim(altText.asText());
}
```

Reply:
I don't think so. Per the old logic, a fallback should be applied if alt text is not defined. Per the new logic, an empty string should be returned. And we select the logic based on block type.
The [RFC](https://jira.squarespace.net/browse/RFC-277) provides the context, but happy to explain further if needed.

Reply:
Hmm I assumed since this code is inside `ImageFormatter.getAltText` we're already in an image-centric context.

Reply:
Ahh ok I follow the logic now. I recommend that the `blockType` enum be added to the repository and compared by integer value, since it is stable and avoids the string operations.

Reply:
See the example comparing `recordType` here. The `promotedBlockType` string comparison in the legacy compiler (before my time) and should be avoided in new code:
https://github.com/Squarespace/template-compiler/blob/0a118bbb887b9188a55222323cda2d222c98d433/core/src/main/java/com/squarespace/template/plugins/platform/ContentPredicates.java#L474

Reply:
I added a `BlockType` enum. I kept the deprecated fields to match `cms-legacy-data` as closely as possible, but I can remove them if needed.